pandas/_libs/groupby.pyx

```diff
@@ -483,8 +483,10 @@ def _group_add(complexfloating_t[:, :] out,
         complexfloating_t val, count
         complexfloating_t[:, :] sumx
         int64_t[:, :] nobs
+        Py_ssize_t len_values = len(values)
+        Py_ssize_t len_labels = len(labels)
 
-    if len(values) != len(labels):
+    if len_values != len_labels:
```

Comment:
i think elsewhere we get around this be using `values.shape[0]` instead of `len`.  IIRC this is something we wont need to do anymore in cython3

Reply:
There are indeed a few cases of 

```
    # TODO(cython 3.0):
    # Instead of `labels.shape[0]` use `len(labels)`
    if not len(values) == labels.shape[0]:
```

in this file. But so if we need to update it anyways after cython 3.0, it doesn't maybe matter that much.

Reply:
yeah i think ok for now